### PR TITLE
💻  v2 Skeleton

### LINKED ISSUE #166: Skeleton

### PR DESCRIPTION
**Pull request by jeffzh4ng**
## Screenshots
https://recordit.co/gC6ErxZoLG

## Purpose
Closes #166 

v2 Skeleton: Replacing clubs and events with housing, study spots, and washrooms. The components are now generic "resource" components, which will support all three resources.

### CODE REVIEW

File: client/components/housing/HouseCard.tsx
Comment (xujustinj) [inline client/components/housing/HouseCard.tsx:4]: If we'll be using tags in the future, can we wait until #138 is in before we merge this? The persisting tech debt in this branch is somewhat uncomfortable, and tags is a big part of it

File: client/components/housing/HouseCard.tsx
Comment (xujustinj) [inline client/components/housing/HouseCard.tsx:108]: I think this is the actual correct pathname given what you have in `NavBar.tsx`

```suggestion
    router.push({ pathname: `/housing/${id}` })
```

File: client/components/housing/HouseCard.tsx
Comment (jeffzh4ng) [reply client/components/housing/HouseCard.tsx:4]: Merged in #138

File: client/components/housing/HouseCard.tsx
Comment (jeffzh4ng) [reply client/components/housing/HouseCard.tsx:108]: Good catch, changed.

File: client/components/club-details/ResourceInfo.tsx
Comment (xujustinj) [inline client/components/club-details/ResourceInfo.tsx:82]: If Prettier did this, that's some interesting formatting here

File: client/utils/types.ts
Comment (xujustinj) [inline client/utils/types.ts:19]: you'll hear more about this from Backend later 😄

File: client/utils/types.ts
Comment (xujustinj) [inline client/utils/types.ts:31]: this more realistically would be a userID, I guess this means we'll have users?

**Review by xujustinj — COMMENTED** [review]
Here's why the Vercel preview is failing:

```
01:03:13.364  	Failed to compile.
01:03:13.365  	./components/club-details/EventsHostedByClub.tsx:5:10
01:03:13.365  	Type error: Module '"../../utils"' has no exported member 'BasicEvent'.
01:03:13.365  	  3 | import { TagBubble } from '../common/TagBubble'
01:03:13.365  	  4 | import { colours, desktopFontSize, device, fontWeight, largerThan, mobile } from '../../styles'
01:03:13.365  	> 5 | import { BasicEvent } from '../../utils'
01:03:13.365  	    |          ^
01:03:13.366  	  6 | 
01:03:13.366  	  7 | const EventsTitle = styled.h2`
01:03:13.366  	  8 |   margin-top: 40px;
```

until that's fixed I don't really know what's going on

File: client/utils/types.ts
Comment (jeffzh4ng) [reply client/utils/types.ts:31]: Hmm, depending on how much dev resources we have left...maybe we should just make all comments anonymous for v2?

File: client/components/club-details/ResourceInfo.tsx
Comment (jeffzh4ng) [reply client/components/club-details/ResourceInfo.tsx]: Prettier somehow doesn't catch this — fixed.

**Review by xujustinj — APPROVED** [review]
preview looks good

File: client/components/club-details/ResourceInfo.tsx
Comment (jeffzh4ng) [reply client/components/club-details/ResourceInfo.tsx:93]: Yeah, I was thinking to keep it if we wanted to include some metadata later.

File: client/utils/serde/clubs-and-events.ts
Comment (jeffzh4ng) [reply client/utils/serde/clubs-and-events.ts:5]: Since this PR removes `Clubs` and `Events`, this compiler complains about the whole module. Commenting for now because we still want to add de/serialization to API types once we get that hooked up.

**Pull request merged**
